File: backend/internal/domain/question.go
package domain

type QuestionType string
type SourceType string

const (
	QuestionTypeMultipleChoice QuestionType = "multiple_choice"
	QuestionTypeDescriptive    QuestionType = "descriptive"

	SourceTypeHighlight SourceType = "highlight"
	SourceTypeNote      SourceType = "note"
	SourceTypeManual    SourceType = "manual"
)

type Question struct {
	ID            string
	QuestionType  QuestionType
	Content       string
	Options       []string
	CorrectAnswer string
	Explanation   string
}

func (q *Question) IsCorrect(answer string) bool {
	return q.CorrectAnswer == answer
}

File: frontend/src/types/question.ts
export interface Question {
  id: string;
  question_type: "multiple_choice" | "descriptive";
  content: string;
  options: string[];
  correct_answer: string;
  explanation: string;
}

export interface SaveQuestionResult {
  question_id: string;
  note: string;
  saved: boolean;
}

export interface SavedQuestion {
  id: string;
  question_type: "multiple_choice" | "descriptive";
  content: string;
  options: string[];
  correct_answer: string;
  explanation: string;
  note: string;
  saved_at: string;
}

export interface IncorrectQuestion {
  id: string;
  question_type: "multiple_choice" | "descriptive";
  content: string;
  options: string[];
  correct_answer: string;
  explanation: string;
  note: string;
  answered_at: string;
}

export interface AnswerResult {
  is_correct: boolean;
  correct_answer: string;
  explanation: string;
  score?: number;
  feedback?: string;
}
